src/fidesops/main.py

```diff
@@ -40,6 +42,75 @@
         allow_headers=["*"],
     )
 
+
+@app.middleware("http")
+async def dispatch_log_request(request: Request, call_next: Callable) -> Response:
+    fides_source: Optional[str] = request.headers.get("X-Fides-Source")
+    now: datetime = datetime.now(tz=timezone.utc)
+    endpoint = f"{request.method}: {request.url.path}"
+
+    try:
+        response = await call_next(request)
+        # HTTPExceptions are considered a handled err by default so are not thrown here.
+        # Accepted workaround is to inspect status code of response.
+        # More context- https://github.com/tiangolo/fastapi/issues/1840
+        try:
+            if response.status_code >= 400:
+                response.background = BackgroundTask(
+                    prepare_and_log_request,
+                    endpoint,
+                    response.status_code,
+                    now,
+                    fides_source,
+                    "HTTPException",
+                )
+            else:
+                response.background = BackgroundTask(
+                    prepare_and_log_request,
+                    endpoint,
+                    response.status_code,
+                    now,
+                    fides_source,
+                    None,
+                )
+        except Exception as exc:
+            # always continue if something went wrong with analytics
+            logger.warning(
+                f"Analytics event for endpoint {request.url} failed to send: {exc}"
+            )
+            return response
+
+        return response
+    except Exception as e:
+        logger.warning("exception caught")
+        prepare_and_log_request(
+            endpoint, e.args[0], now, fides_source, e.__class__.__name__
+        )
+        raise
+
+
+def prepare_and_log_request(
+    endpoint: str,
+    status_code: int,
+    event_created_at: datetime,
+    fides_source: Optional[str],
+    error_class: Optional[str],
+) -> None:
+    analytics_event = AnalyticsEvent(
+        docker=in_docker_container(),
+        event=Event.endpoint_call.value,
+        event_created_at=event_created_at,
+        local_host=running_on_local_host(),
```

Comment:
Why is this always true?

Reply:
we're always running on localhost from the perspective of 1. always either running within the docker container or 2. running locally e.g. using pyenv or something. In this way, all ports/addresses are referencing internal/local addresses.